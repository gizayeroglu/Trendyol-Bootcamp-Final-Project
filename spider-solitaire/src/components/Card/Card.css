:root {
  --white: #e8e8e6;
  --black: #000;
}

.card {
  height: 130px;
  width: 95px;
  border-radius: 10px;
  border: none;
  box-shadow: 0 -2px 5px 0 rgba(0, 0, 0, 0.8);
  margin-top: -95px;
}
.card-back {
  background: url("../../assets/spider-solitaire-card-back.webp");
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center center;
  display: block !important;
}
.card-back > span {
  display: none;
}
.card-front {
  background-color: var(--white);
  background: url("../../assets/spider-solitaire-club-card.webp") var(--white);
  color: var(--black);
  background-repeat: no-repeat;
  background-position: center;
  background-size: 30%;
  position: relative;
  cursor: pointer;
}
.card-front:active {
  box-shadow: 0 5px 10px 5px rgba(0, 0, 0, 0.6);
}
.card-front .content-left-top {
  display: flex;
  flex-direction: column;
  align-items: center;
  top: 2px;
  left: 5px;
  position: absolute;
  font-size: 17px;
  font-weight: 700;
}
.card-front .content-right-bottom {
  display: flex;
  flex-direction: column;
  align-items: center;
  bottom: 2px;
  right: 5px;
  position: absolute;
  font-size: 17px;
  font-weight: 700;
  transform:rotate(180deg);
}
:is(.content-left-top, .content-right-bottom) img {
  width: 13px;
}