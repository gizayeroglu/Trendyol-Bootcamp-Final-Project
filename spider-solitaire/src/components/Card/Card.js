import './Card.css';

import React from 'react';
import cardImage from '../../assets/club.png';

function Card({symbol, isOpen, onDragStart, dragging}){
  let classes = isOpen ? `card card-front` : `card card-back`;

  return(
    <>
      <div onDragStart={onDragStart} onDrag={dragging} draggable={isOpen} className={classes}>
        <span className='content-left-top'>{symbol}<img src={cardImage} alt='club'/></span>
        <span className='content-right-bottom'>{symbol}<img src={cardImage} alt='club'/></span>
      </div>
    </>
  )
}

export default Card;