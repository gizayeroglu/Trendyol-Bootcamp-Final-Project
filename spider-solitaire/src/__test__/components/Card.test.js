import React from "react";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { configure, mount, shallow } from "enzyme";

import Card from "../../components/Card/Card";

configure({ adapter: new Adapter() });

describe("Card component test", function () {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Card />);
  });

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should display card as open if is last card props is true", () => {
    const isLastCard = true;
    const component = shallow(<Card isLastCard={isLastCard} />);
    let card = component.find("div");
    expect(card.hasClass("card-front")).toBeTruthy();
  });

  it("should display card as closed if is last card props is false", () => {
    const isLastCard = false;
    const component = shallow(<Card isLastCard={isLastCard} />);
    let card = component.find("div");
    expect(card.hasClass("card-back")).toBeTruthy();
  });
});
