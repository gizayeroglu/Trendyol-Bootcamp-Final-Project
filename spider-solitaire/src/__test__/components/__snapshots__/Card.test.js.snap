// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component test should render 1`] = `ShallowWrapper {}`;
